baseapp/server.go

```diff
@@ -33,6 +34,7 @@ type Server struct {
 	middleware []func(http.Handler) http.Handler
 	logger     zerolog.Logger
 	mux        *goji.Mux
+	server     *http.Server
```

Comment:
@bluekeyes I've exposed the server object here to allow for updates to it later

Reply:
Thanks. As this is not an exported field, we'll also need a method to access it. I can think of two approaches right now:

1. Define a `Server() *http.Server` method. Simple, but might imply that calling `ListenAndServe` directly on the returned value is a good idea. Also requires callers to create the `baseapp.Server` first and then manipulated it.

2. Define a `WithServer(*http.Server)` option function that allows the user to provide their own `http.Server` instance. If not provided, we use a default and otherwise we set any values (like the address) that are unset and use it.

I like (2), as it matches how we've configured other things like the metrics registry.